### PR TITLE
[Automation] Bootstrap with custom ssh and support for manage services

### PR DESCRIPTION
**Pull request by pranavprakash20**
Signed-off-by: Pranav <prprakas@redhat.com>

TC's included
CEPH-83575205 Bootstrap a cluster using cephadm-ansible module 'cephadm_bootstrap' with overriding existing ceph keys
CEPH-83575215 Get 'mgr' config using cephadm-ansible module 'ceph_config'
CEPH-83575214 Set 'osd' config using cephadm-ansible module 'ceph_config'

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Run log :  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DBXT8R/

File: playbooks/bootstrap-cluster-custom-ssh.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+
+- name: Test 'cephadm_bootstrap' module
+  hosts: admin
+  gather_facts: false
+  become: true
+  any_errors_fatal: true
+
+  tasks:
+    - name: Bootstrap with dashboard set to `true` without credenatials
```
Comment (vamahaja) [inline playbooks/bootstrap-cluster-custom-ssh.yml:10]: Change `name` accordinlgy

File: playbooks/bootstrap-cluster-custom-ssh.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      cephadm_bootstrap:
+        mon_ip: "{{ mon_ip }}"
+        dashboard: true
+        monitoring: false
+        firewalld: false
+        ssh_user: ceph-admin
```
Comment (vamahaja) [inline playbooks/bootstrap-cluster-custom-ssh.yml:16]: is `ceph-admin` user available in deployed cluster ?

File: tests/cephadm/test_cephadm_ansible_manage_services.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from os.path import basename
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_ansible_manage_services.py:1]: This looks similar to [1], try to utilize same test module.
[1] https://github.com/red-hat-storage/cephci/pull/2049

File: playbooks/restart-mgr-service.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---
+
+- name: Restart mon.0 service
+  hosts: bootstrap
+  gather_facts: false
+  become: true
+  any_errors_fatal: true
+  tasks:
+    - name:
```
Comment (vamahaja) [inline playbooks/restart-mgr-service.yml:9]: missing name

File: playbooks/restart-mgr-service.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      ceph_orch_daemon:
+        state: restarted
+        daemon_id: mgr.0
+        daemon_type: mgr
```
Comment (vamahaja) [inline playbooks/restart-mgr-service.yml:13]: Better to add validation step for restarted service.

File: suites/pacific/cephadm/test_cephadm_bootstrap_custom_ssh.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+#    conf/pacific/upgrades/tier-1_upgrade_cephadm.yaml
+#
+#    4-Node cluster(RHEL-8.3 and above)
+#    3 MONS, 2 MDS, 1 MGR, 3 OSD and 2 RGW service daemon(s)
+#     Node1 - Mon, Mgr, Installer, OSD, alertmanager, grafana, prometheus, node-exporter
+#     Node2 - Mon, Mgr, OSD, MDS, RGW, alertmanager, node-exporter
+#     Node3 - Mon, OSD, MDS, RGW, node-exporter
+#     Node4 - Client
+#
+# Tc's covered:
```
Comment (vamahaja) [inline suites/pacific/cephadm/test_cephadm_bootstrap_custom_ssh.yaml:16]: Not required, it's already part of test description

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: playbooks/bootstrap-cluster-custom-ssh.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  any_errors_fatal: true
+
+  tasks:
+    - name: Bootstrap with dashboard set to `true` without credenatials
+      cephadm_bootstrap:
+        mon_ip: "{{ mon_ip }}"
+        dashboard: true
+        monitoring: false
+        firewalld: false
+        ssh_user: cephuser
```
Comment (vamahaja) [inline playbooks/bootstrap-cluster-custom-ssh.yml:16]: Will user `cephuser` available in cluster ?

Comment (pranavprakash20) [reply playbooks/bootstrap-cluster-custom-ssh.yml:16]: Yes Vaibhav, its available in https://github.com/red-hat-storage/cephci/blob/master/conf/inventory/rhel-8.7-server-x86_64.yaml#L15

Comment (vamahaja) [reply playbooks/bootstrap-cluster-custom-ssh.yml:16]: +1

**Review by vamahaja — APPROVED** [review]

**Review by MohitBis — APPROVED** [review]
LGTM

**Pull request merged**